File: DoLearn.API/Models/Course.cs
using DoLearn.API.Models;

public class Course
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required string CourseCode { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime LastUpdated { get; set; } = DateTime.UtcNow;

    public int CreatedById { get; set; }
    public string? ImgURL { get; set; }

    public User CreatedBy { get; set; } = null!;

    public CoursePricing? Pricing { get; set; }
    public CourseSchedule Schedule { get; set; } = null!;

    public List<User> Students { get; set; } = [];
    public List<User> Teachers { get; set; } = [];
}

File: do-learn-frontend/src/app/models/Course.ts
export interface Course {
    id: number;
    title: string;
    description: string;
    courseCode: string;
    startDate: string;
    endDate: string;
    imgIRL: string;
    createdAt: string;       // Changed from createdById
    lastUpdated: string;     // Changed from lastUpdated
    imgURL: string | null;   // Corrected from imgIRL
  }
  